### PR TITLE
SB-22541 : CourseBatch date changes updation

### PR DESCRIPTION
**Pull request by neha0305verma**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-22541" title="SB-22541" target="_blank">StorySB-22541</a>  MC - Certificates enhancements
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Java-11, play2-2.7.2, scala-2.11, redis-5.0.3
* Hardware versions: 2 CPU / 4GB RAM

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/JsonKey.java [hunk 1]
```diff
@@ -1044,9 +1044,10 @@
   public static final String X_DEVICE_ID = "x-device-id";
   public static final String X_SESSION_ID = "x-session-id";
   public static final String X_TRACE_ID = "x-trace-id";
-  public static final String CONVERT_DATE_AS_STRING = "convertDateAsString";
   public static final String USER_ENROLMENTS_DB = "user_enrolments";
-  public static final List<String> CHANGE_IN_DATE_FORMAT = new ArrayList<String>() {{add("startDate"); add("endDate"); add("enrollmentEndDate");}};
+  public static final List<String> CHANGE_IN_SIMPLE_DATE_FORMAT = new ArrayList<String>() {{add("startDate"); add("endDate"); add("enrollmentEndDate");}};
+  public static final List<String> CHANGE_IN_DATE_FORMAT = new ArrayList<String>() {{add("createdDate"); add("updatedDate");}};
+  public static final List<String> CHANGE_IN_DATE_FORMAT_ALL = new ArrayList<String>() {{add("startDate"); add("endDate"); add("enrollmentEndDate"); add("createdDate"); add("updatedDate");}};
```
Comment (PradyumnaNagendra) [inline course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/JsonKey.java:1050]: you can use Arrays.asList

Comment (PradyumnaNagendra) [reply course-mw/sunbird-util/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/models/util/JsonKey.java:1050]: do the same for this also 
`public static final List<String> SET_END_OF_DAY = new ArrayList<String>() {{add("endDate"); ad("enrollmentEndDate");}};`

**Review by PradyumnaNagendra — CHANGES_REQUESTED** [review]

**Review by PradyumnaNagendra — APPROVED** [review]

File: course-mw/course-actors-common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java [hunk 2]
```diff
@@ -34,6 +41,17 @@
   private static ElasticSearchService esUtil = EsClientFactory.getInstance(JsonKey.REST);
   private static ObjectMapper mapper = new ObjectMapper();
   private static LoggerUtil logger = new LoggerUtil(CourseBatchUtil.class);
+  private static final SimpleDateFormat DATE_FORMAT = ProjectUtil.getDateFormatter();
+  private static final SimpleDateFormat DATE_FORMAT_SIMPLE = new SimpleDateFormat("yyyy-MM-dd");
```
Comment (maheshkumargangula) [inline course-mw/course-actors-common/src/main/java/org/sunbird/learner/util/CourseBatchUtil.java:45]: SimpleDateFormat - Concurrency will create issues for the format. Please use it as local variable.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java [hunk 3]
```diff
@@ -328,30 +325,37 @@ private void validateMentors(CourseBatch courseBatch, String authToken, RequestC
...
         Map<String, Map<String, Object>> mentorDetails =
                 mentorDetailList.stream().collect(Collectors.toMap(map -> (String) map.get(JsonKey.ID), map -> map));
-        
-        for(String mentorId: mentors) {
+
+        for (String mentorId : mentors) {
           Map<String, Object> result = mentorDetails.getOrDefault(mentorId, new HashMap<>());
-          if(MapUtils.isEmpty(result) || (result.containsKey(JsonKey.IS_DELETED) && (Boolean) result.get(JsonKey.IS_DELETED))) {
+          try {
+            if (MapUtils.isEmpty(result) || (result.containsKey(JsonKey.IS_DELETED) && (Boolean) result.get(JsonKey.IS_DELETED))) {
```
Comment (PradyumnaNagendra) [inline course-mw/course-actors/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchManagementActor.java:335]: result.get(JsonKey.IS_DELETED, false)

**Review by PradyumnaNagendra — CHANGES_REQUESTED** [review]

**Review by PradyumnaNagendra — APPROVED** [review]

**Review by maheshkumargangula — APPROVED** [review]

**Pull request merged**